DOCSP-33788-migration-indexes

ltran-mdb2 opened the pull request:
## DESCRIPTION

Adds information about indexes created/not created during migration. 

## STAGING

https://deploy-preview-177--docs-relational-migrator.netlify.app/jobs/sync-jobs/#indexes

## JIRA

https://jira.mongodb.org/browse/DOCSP-33788

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Is this free of spelling errors?
- [x] Is this free of grammatical errors?
- [x] Is this free of staging / rendering issues?
- [x] Are all the links working?

## External Review Requirements

[What's expected of an external reviewer?](https://wiki.corp.mongodb.com/display/DE/Reviewing+Guidelines+for+the+MongoDB+Server+Documentation)

sarah-olson-mongodb commented on source/jobs/sync-jobs.txt:
```diff
@@ -54,6 +54,18 @@ To enable idempotent inserts for documents and embedded arrays, set
+
+Relational Migrator does not create new indexes in the relational
+database. 
+
```
(suggestion) 
It might be helpful to link out to the [Indexes](https://www.mongodb.com/docs/manual/indexes/) section in the manual. 

It might also be helpful to link specifically to the create, drop, and get indexes reference pages.

ltran-mdb2 replied:
Done!

sarah-olson-mongodb commented on source/jobs/sync-jobs.txt:
```diff
@@ -54,6 +54,18 @@ To enable idempotent inserts for documents and embedded arrays, set
+Indexes
+~~~~~~~
+
+Relational Migrator creates new indexes in your MongoDB collection to
```
(questions)
Does this happen automatically, with every job? 
Is it useful to state when in the sync job process this happens?

ltran-mdb2 replied:
@tomhollander  could you clarify here?

davidsoulmdb replied:
Correct, RM will automatically create some basic additional indexes (beyond the built-in _id index) when relevant to a project mapping, eg when we have arrays and need to update the array contents.

ltran-mdb2 replied:
@davidsoulmdb Do you think it is useful to state when in the sync job process this happens?

ltran-mdb2 replied:
Discussed over slack and determined that this information is not useful for users to know.

sarah-olson-mongodb reviewed (approved):
Thanks so much @ltran-mdb2! This LGTM with a few suggestions / questions.

tomhollander commented:
Can we get @davidsoulmdb and @alwaysbemark to review please?

davidsoulmdb commented:
Overall great, a few optional suggestions;
1. Relational Migrator does not create new indexes in the relational database. -> Relational migrator does not create any new indexes in the source database, as it is designed to be run with a read-only access to the source database. We recommend that you configure appropriate source database indexes to process SQL queries for database ids and table filter clauses.
2. Unrelated; can we mention that snapshot jobs use table filters if present, but CDC jobs always ignore snapshot table filters?
3. Relational Migrator creates new indexes in your MongoDB collection -> "Relational Migrator creates indexes in your MongoDB collections" plural. 
4. I would recommend something like "Administrators will general add multiple additional indexes to MongoDB to fit their application usage patterns."
5. Should this be a checklist/numbered list?

ltran-mdb2 commented:
Thanks for your feedback @davidsoulmdb ! Could you take another look when you get a chance? 

> Overall great, a few optional suggestions;
> 
> 1. Relational Migrator does not create new indexes in the relational database. -> Relational migrator does not create any new indexes in the source database, as it is designed to be run with a read-only access to the source database. We recommend that you configure appropriate source database indexes to process SQL queries for database ids and table filter clauses - we could potentially add this information to the [prerequisites](https://www.mongodb.com/docs/relational-migrator/jobs/prerequisites/my-sql/) page? 

I added the read-only information but I'm hesitant to add the recommendation because I'm not sure it fits in the context of this page. 

> 2. Unrelated; can we mention that snapshot jobs use table filters if present, but CDC jobs always ignore snapshot table filters?

Would you mind opening another ticket to address this just to maintain scope? 
> 3. Relational Migrator creates new indexes in your MongoDB collection -> "Relational Migrator creates indexes in your MongoDB collections" plural.

Done!

> 4. I would recommend something like "Administrators will general add multiple additional indexes to MongoDB to fit their application usage patterns."

I changed the wording to include this!

> 5. Should this be a checklist/numbered list?

Because these are behavioral considerations as opposed to sequential information, I'm hesitant to made this info a checklist or numbered list

davidsoulmdb reviewed (approved):
Thanks, LGTM and I'll parse the 2 other suggestions as separate tickets too.

davidsoulmdb commented:
FYI table filters ticket at https://jira.mongodb.org/browse/DOCSP-45462 and wont push on the read only suggestion.
